### ISSUE TITLE
`NO_CORRESPONDING_PROPERTY` doesn't seem to be correct

### ISSUE THREAD

**Issue by xiaoqian0408** [saveourtool/diktat#1708]
## Steps to Reproduce
I have a class 

```kotlin
class MyClass constructor(
    private val context: CoroutineContext,
) : Closeable, CoroutineScope {
    override val coroutineContext: CoroutineContext
        get() {
            ...//some code

            return context
        }

    override fun close() {
        coroutineContext.cancel()

        ... //some code
    }

    companion object {
        private const val TAG = "MyClass"
    }
}
```

The warning says 
[`NO_CORRESPONDING_PROPERTY`] backing property should have the same name, but there is no corresponding property: coroutineContext has no corresponding property with name `_coroutineContext` (cannot be auto-corrected) (diktat-ruleset:implicit-backing-property)

But actually inside CoroutineScope interface (in org.jetbrains.kotlinx:kotlinx-coroutines-android, 1.6.4), coroutineContext exists.

## Environment information
* diktat version: org.cqfn.diktat.diktat-gradle-plugin:**1.2.5**
* build tool (maven/gradle): gradle
* how is diktat run (CLI, plugin, etc.): command line
* kotlin version: 1.8.10
* operating system: Mac OS
* link to a project (if your project is public): sorry I can't, it's private, but I pasted my sample code above

**Comment by 0x6675636b796f75676974687562** [comment]
@xiaoqian0408, thank you for your report.

I'm looking into the issue and will get back to you in a short while.

**Comment by 0x6675636b796f75676974687562** [comment]
@xiaoqian0408, _Diktat_ warns you not about the `coroutineContext` property, but about the name of the private `context` property, which is a [_backing field_](https://kotlinlang.org/docs/properties.html#backing-properties) and therefore should be named `_coroutineContext`.

You can rewrite your code as follows:

```kotlin
import java.io.Closeable
import kotlin.coroutines.CoroutineContext
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.cancel

class MyClass constructor(
    context: CoroutineContext,
) : Closeable, CoroutineScope {
    @Suppress("VARIABLE_NAME_INCORRECT_FORMAT")
    private val _coroutineContext: CoroutineContext = context

    override val coroutineContext: CoroutineContext
        get() {
            // Custom code here
            return _coroutineContext
        }

    override fun close() {
        // Custom code here
        coroutineContext.cancel()
    }

    companion object {
        private const val TAG = "MyClass"
    }
}
```

Unfortunately, _Diktat_ doesn't have full support for _backing fields_ as of yet (feel free to upvote #1711), so you'll have to suppress the `VARIABLE_NAME_INCORRECT_FORMAT` false positive.

You can simplify the code even further, however:

```kotlin
import java.io.Closeable
import kotlin.coroutines.CoroutineContext
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.cancel

class MyClass constructor(
    context: CoroutineContext,
) : Closeable, CoroutineScope {
    override val coroutineContext: CoroutineContext = context
        get() {
            // Custom code here
            return field
        }

    override fun close() {
        // Custom code here
        coroutineContext.cancel()
    }

    companion object {
        private const val TAG = "MyClass"
    }
}
```

This way, _Diktat_ will issue no warnings at all.

**Comment by akuleshov7** [comment]
Let's finish it in #1711 . As described by @0x6675636b796f75676974687562 you can have workaround he proposed or even suppress this rule with `@Suppress("VARIABLE_NAME_INCORRECT_FORMAT")`
